DEVDOCS-2528: Embedded Checkout Domain Setup (#970)
bigcommerce/dev-docs: docs/api-docs/cart-and-checkout/embedded-checkout-overview.md

- [Channels, Sites, and Routes APIs](#channels-sites-and-routes-apis)
- [BigCommerce Checkout SDK](#bigcommerce-checkout-sdk)
- [Logged-In customers](#logged-in-customers)
- [FAQ](#faq)

</div>

> Embedded Checkout lets you place BigCommerce's checkout onto any website. Customers can check out on an external storefront while their order information syncs simultaneously to the BigCommerce Control Panel. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for Wordpress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).

Reviewer: ```suggestion
Embedded Checkout lets you place BigCommerce's checkout onto any website. Customers can check out on an external storefront while their order information syncs simultaneously to the BigCommerce Control Panel. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for WordPress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).
```

> Embedded Checkout lets you place BigCommerce's checkout onto any website. Customers can check out on an external storefront while their order information syncs simultaneously to the BigCommerce Control Panel. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for Wordpress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).

Reviewer 2: Embedded Checkout lets you place BigCommerce's checkout onto any website. Customers can check out on an external storefront while their order information syncs simultaneously to the BigCommerce control panel. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for Wordpress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).

> Embedded Checkout lets you place BigCommerce's checkout onto any website. Customers can check out on an external storefront while their order information syncs simultaneously to the BigCommerce Control Panel. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for WordPress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).

Reviewer 3: @aglensmith, the statements made here and the reference to BC4WP may no longer be true.
See https://bigcommerce.slack.com/archives/C57SQ401K/p1614112903001300.

> The [Checkout SDK's](https://github.com/bigcommerce/checkout-sdk-js) [`embedded-checkout` sub-module](https://github.com/bigcommerce/checkout-sdk-js/blob/master/docs/README.md#embedcheckout) can be used to embed [BigCommerce's Optimized One-Page Checkout](https://support.bigcommerce.com/s/article/Optimized-Single-Page-Checkout) into non-native storefronts like WordPress. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for Wordpress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).

Reviewer 3: ```suggestion
The [Checkout SDK's`embedded-checkout` sub-module](https://github.com/bigcommerce/checkout-sdk-js/blob/master/docs/README.md#embedcheckout) can be used to embed [BigCommerce's Optimized One-Page Checkout](https://support.bigcommerce.com/s/article/Optimized-Single-Page-Checkout) into non-native storefronts like WordPress. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. For more information about the plugin, see [BigCommerce for Wordpress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).
```

## How it works

Embedded Checkout uses an HTML `<iframe>` to display BigCommerce's PCI compliant Optimized One-Page Checkout on non-BigCommerce web pages.

If your channel site doesn't match the URL from which you're making a request to a BigCommerce store, you will get a security error and the checkout will not load. Additionally, if requests to your BigCommerce store aren't served over HTTPS, you will also see an error.

## Channels, Sites, and Routes APIs

> You will need to use the [Channels, Sites, and Routes](https://developer.bigcommerce.com/api-reference/cart-checkout/channels-listings-api) APIs to embed checkout on an external site. The Channels API allows you to create and manage sales channel listings across a merchant's product catalog. A channel can be a marketplace, like Amazon, or an external storefront, like a WordPress site. The Sites and Routes APIs let you set an external storefront domain and define the paths for important pages, like the home page, cart page, or checkout page. The Sites and Routes APIs allow you to link back to the proper URL from invoice emails and storefront links.

Reviewer: ```suggestion
You will need to use the [Channels API](https://developer.bigcommerce.com/api-reference/store-management/channels) and the [Sites and Routes API](https://developer.bigcommerce.com/api-reference/store-management/sites) to embed checkout on an external site. The Channels API allows you to create and manage sales channel listings across a merchant's product catalog. A channel can be a marketplace, like Amazon, or an external storefront, like a WordPress site. The Sites and Routes API lets you set an external storefront domain and define the paths for important pages, like the home page, cart page, or checkout page. The Sites and Routes API allows you to link back to the proper URL from invoice emails and storefront links.
```

## BigCommerce Checkout SDK

> Embedded Checkout requires the BigCommerce Checkout SDK to invoke a method to render the checkout in your site. Learn more about the [Checkout SDK](https://developer.bigcommerce.com/api-docs/cart-and-checkout/checkout-sdk).

Reviewer: ```suggestion
Embedded Checkout requires the BigCommerce [Checkout SDK](https://developer.bigcommerce.com/api-docs/cart-and-checkout/checkout-sdk) to invoke a method to render the checkout in your site.
```

> ## Logged-In customers

Reviewer 2: ## Logged-in customers

> The [Customer Login API](https://developer.bigcommerce.com/api-docs/customers/customer-login-api) allows you to manage customers in two steps. First, you need to pass the `customer_id` when creating the cart. Second, you need to log in the customer, so the session is active when the checkout loads.

Reviewer: ```suggestion
The [Customer Login API](https://developer.bigcommerce.com/api-docs/customers/customer-login-api) allows you to manage customers in two steps. First, you need to pass the `customer_id` when creating the cart. Second, you need to log the customer in, so the session is active when the checkout loads.
```

> You will need to use the [Channels API](https://developer.bigcommerce.com/api-reference/store-management/channels) and the [Sites and Routes API](https://developer.bigcommerce.com/api-reference/store-management/sites) to embed checkout on an external site. The Channels API allows you to create and manage sales channel listings across a merchant's product catalog. A channel can be a marketplace, like Amazon, or an external storefront, like a WordPress site. The Sites and Routes API lets you set an external storefront domain and define the paths for important pages, like the home page, cart page, or checkout page. The Sites and Routes API allows you to link back to the proper URL from invoice emails and storefront links.

Reviewer 3: ```suggestion
You will need to use the [Channels API](https://developer.bigcommerce.com/api-reference/store-management/channels) and the [Sites and Routes API](https://developer.bigcommerce.com/api-reference/store-management/sites) to embed a checkout on an external site. The Channels API allows you to create and manage sales channel listings across a merchant's product catalog. A channel can be a marketplace, like Amazon, or an external storefront, like a WordPress site. The Sites and Routes API lets you set an external storefront domain and define the paths for important pages, like the home page, cart page, or checkout page. The Sites and Routes API allows you to link back to the proper URL from invoice emails and storefront links.
```

## FAQ

> ### How do I resolve Embedded Checkout 403 "Cannot start checkout session with an empty cart" Errors?**

Reviewer: ```suggestion
### How do I resolve Embedded Checkout 403 "Cannot start checkout session with an empty cart" Error?**
```

> ### How do I resolve Embedded Checkout 403 "Cannot start checkout session with an empty cart" Errors?**

Reviewer 3: ```suggestion
### How do I resolve Embedded Checkout 403 "Cannot start checkout session with an empty cart" Errors?
```

> In order for Embedded Checkout to work correctly for shoppers using a browser with restricted privacy settings (like Apple's Safari), your checkout page must be served from the same domain as your BigCommerce storefront. For example, if your headless storefront is `www.mystore.com`, then your BigCommerce store's domain should be `checkout.mystore.com`. See [BigCommerce for WordPress](https://support.bigcommerce.com/s/article/BigCommerce-for-WordPress-Checkout#safari) in the Help Center for more information on making Embedded Checkout on a headless WordPress storefront compatibile with Safari.

Reviewer: ```suggestion
For Embedded Checkout to work correctly for shoppers using a browser with restricted privacy settings (like Apple's Safari), your checkout page must be served from the same domain as your BigCommerce storefront. For example, if your headless storefront is `www.mystore.com`, then your BigCommerce store's domain should be `checkout.mystore.com`. For more information on making Embedded Checkout on a headless WordPress storefront compatible with Safari, see [BigCommerce for WordPress](https://support.bigcommerce.com/s/article/BigCommerce-for-WordPress-Checkout#safari) in the Help Center. 

```

> In order for Embedded Checkout to work correctly for shoppers using a browser with restricted privacy settings (like Apple's Safari), your checkout page must be served from the same domain as your BigCommerce storefront. For example, if your headless storefront is `www.mystore.com`, then your BigCommerce store's domain should be `checkout.mystore.com`. See [BigCommerce for WordPress](https://support.bigcommerce.com/s/article/BigCommerce-for-WordPress-Checkout#safari) in the Help Center for more information on making Embedded Checkout on a headless WordPress storefront compatibile with Safari.

Reviewer 2: For Embedded Checkout to work correctly for shoppers using a browser with restricted privacy settings (like Apple's Safari), you must serve your checkout page from the same domain as your BigCommerce storefront. For example, if your headless storefront is `www.mystore.com`, your BigCommerce store's domain should be `checkout.mystore.com`. See [BigCommerce for WordPress](https://support.bigcommerce.com/s/article/BigCommerce-for-WordPress-Checkout#safari) in the Help Center for more information on making Embedded Checkout on a headless WordPress storefront compatible with Safari.

> For Embedded Checkout to work correctly for shoppers using a browser with restricted privacy settings (like Apple's Safari), your checkout page must be served from the same domain as your BigCommerce storefront. For example, if your headless storefront is `www.mystore.com`, then your BigCommerce store's domain should be `checkout.mystore.com`. For more information on making Embedded Checkout on a headless WordPress storefront compatible with Safari, see [BigCommerce for WordPress](https://support.bigcommerce.com/s/article/BigCommerce-for-WordPress-Checkout#safari) in the Help Center.

Reviewer 3: ```suggestion
For Embedded Checkout to work correctly for shoppers using a browser with restricted privacy settings (like Apple's Safari), your checkout page must be served from the same domain as your BigCommerce storefront. For example, if your headless storefront is `www.mystore.com`, then your BigCommerce store's domain should be `checkout.mystore.com`. For more information on making Embedded Checkout on a headless WordPress storefront compatible with Safari, see [BigCommerce for WordPress](https://support.bigcommerce.com/s/article/BigCommerce-for-WordPress-Checkout#safari).
```
